rfcs/0005-fork-selection.md

```diff
@@ -0,0 +1,99 @@
+# Fork Selection
+ This RFC defines Solana's *Nakomoto Fork Selection* algorithm based on time locks. It satisfies the following properties:
```

Comment:
Can you remove any references to RFC? And move this to `book/src/fork-selection.md`? And reference it in `book/src/SUMMARY.md` under "Proposed Architectural Changes"?